src/components/simple-color-picker/simple-color-picker.component.tsx

```diff
@@ -41,21 +68,53 @@ const SimpleColorPicker = (props) => {
     ...rest
   } = props;
 
-  const filteredChildren = useMemo(() => React.Children.toArray(children), [
-    children,
-  ]);
+  const filteredChildren = useMemo(
+    () =>
+      React.Children.toArray(children).filter((child) =>
+        React.isValidElement(child)
+      ) as React.FunctionComponentElement<
+        SimpleColorProps & { ref: React.RefObject<HTMLInputElement> }
+      >[],
+    [children]
+  );
```

Comment:
To be honest I don't really see any advantage of using your proposal over mine

Reply:
I actually did it the same way as you in `AccordionGroup` long time ago, and I think that solution is not right, as we're only forcing a type here, but still that array could contain elements other than `SimpleColor`. I think that you should at least move this code below the invariant error, so it would be triggered before the array is set.

You should also remove the ref part, as `FunctionComponentElement` already has that prop in it's definition.